test/test.js

```diff
@@ -10,22 +10,36 @@ var fs = require( 'fs' ),
 	profiles = require( '../package.json' ).files
 		// Trim ".json" from fileName end
 		.map( fileName => fileName.slice( 0, -5 ) )
-		// TODO: Test language profiles too
-		.filter( fileName => ( fileName.indexOf( 'language/' ) === -1 ) );
+		.filter( fileName => (
+			// TODO: Test language profiles too
+			fileName.indexOf( 'language/' ) === -1 &&
+			// Node rules are tested through server profile
+			fileName.indexOf( 'node' ) === -1
+		) );
 
 profiles.forEach( function ( profile ) {
-	var count, config;
+	var count, config, rules;
 
 	console.log( 'Testing the "' + profile + '" profile suite.' );
 
 	config = require( '../' + profile + '.json' );
 	validFixturesFile = __dirname + '/fixtures/' + profile + '/valid.js';
 	invalidFixturesFile = __dirname + '/fixtures/' + profile + '/invalid.js';
+	rules = config.rules || {};
+
+	if ( profile === 'server' ) {
```

Comment:
Hm.. I'm probably missing something, but why is this needed? I'd expect that ESLint can read the server.json file as-is from the above <profile>.json require. If not, and we need a hardcoded list of rules (which indeed line 42 indicates), then why is there not an exiting block for the client profile?

Reply:
Because server.json doesn't contain rules, it's just common+node+es6. We don't want to re-test common rules here. I considered moving the test fixtures to node/, but what we are actually testing is node+es6.

Arguably we should introduce the same abstraction for client, as one may want to use something other than ES5.

Reply:
ping @Krinkle